torchvision/datasets/mnist.py

```diff
@@ -220,30 +220,23 @@ def download(self):
         if self._check_exists():
             return
 
-        # download files
-        try:
```

Comment:
I think it's better to keep this way of handling existing dirs, or maybe factor this out in a helper function.